core/src/main/java/com/datastax/oss/driver/internal/core/loadbalancing/DefaultLoadBalancingPolicy.java

```diff
@@ -64,8 +64,11 @@
   private volatile DistanceReporter distanceReporter;
   @VisibleForTesting volatile String localDc;
 
-  public DefaultLoadBalancingPolicy(DriverContext context) {
-    this(getLocalDcFromConfig(context), getFilterFromConfig(context), context);
+  public DefaultLoadBalancingPolicy(DriverContext context, String profileName) {
```

Comment:
Small slightly unrelated issue with `DefaultLoadBalancingPolicy`, the logic at line 113 assumes that if no contact points are provided that the localDC be inferred from 127.0.0.1.  We should probably get rid of that logic if the localDC is explicitly set.  In my case I had a default profile using `dc1`, and another profile using `dc2`, and in both cases `dc1` was chosen because no contact points were provided.

Reply:
To clarify, i'm suggesting changing:

```
    if (contactPoints.isEmpty()) {
```

to

```java
    if (localDc == null && contactPoints.isEmpty()) {
```

Reply:
went ahead and pushed d62853d which does that

Reply:
Agreed 👍 If there are no contact points, the local DC is optional, but it can still be specified.
The code can be improved further for readability because we check `localDc!=null` in the next branch, I'm pushing a small commit to change it.